src/Analyzers/Core/Analyzers/UseInterpolatedString/AbstractUseInterpolatedStringDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,66 @@
+    : AbstractBuiltInCodeStyleDiagnosticAnalyzer
+    where TSyntaxKind : struct
+    where TExpressionSyntax : SyntaxNode
+    where TStringLiteralExpressionSyntax : TExpressionSyntax
+{
+    protected AbstractUseInterpolatedStringDiagnosticAnalyzer()
+        : base(IDEDiagnosticIds.UseInterpolatedStringDiagnosticId,
+                EnforceOnBuildValues.UseInterpolatedString,
+                CodeStyleOptions2.PreferInterpolatedString,
+                new LocalizableResourceString(nameof(AnalyzersResources.Use_interpolated_string), AnalyzersResources.ResourceManager, typeof(AnalyzersResources)),
+                new LocalizableResourceString(nameof(AnalyzersResources.String_can_be_converted_to_interpolated_string), AnalyzersResources.ResourceManager, typeof(AnalyzersResources)))
+    {
+    }
+
+    private static readonly LocalizableResourceString s_localizableUseInterpolatedString = new(
+       nameof(AnalyzersResources.Use_interpolated_string), AnalyzersResources.ResourceManager, typeof(AnalyzersResources));
+
+    private static readonly LocalizableResourceString s_localizableStringCanBeConvertedToInterpolatedString = new(
+       nameof(AnalyzersResources.String_can_be_converted_to_interpolated_string), AnalyzersResources.ResourceManager, typeof(AnalyzersResources));
+
+    private static DiagnosticDescriptor CreateDescriptor()
+        => CreateDescriptorWithId(
+            IDEDiagnosticIds.UseInterpolatedStringDiagnosticId,
+            EnforceOnBuildValues.UseInterpolatedString,
+            hasAnyCodeStyleOption: true,
+            s_localizableUseInterpolatedString,
+            s_localizableStringCanBeConvertedToInterpolatedString);
+
+    public sealed override DiagnosticAnalyzerCategory GetAnalyzerCategory()
+        => DiagnosticAnalyzerCategory.SemanticSpanAnalysis;
+
+    protected abstract ISyntaxFacts GetSyntaxFacts();
+
+    protected sealed override void InitializeWorker(AnalysisContext context)
+    {
+        var syntaxFacts = GetSyntaxFacts();
+        var syntaxKinds = syntaxFacts.SyntaxKinds;
+        context.RegisterSyntaxNodeAction(
+            AnalyzeSyntax, syntaxKinds.Convert<TSyntaxKind>(syntaxKinds.StringLiteralExpression));
+    }
+
+    private void AnalyzeSyntax(SyntaxNodeAnalysisContext context)
+    {
+        var stringLiteralExpression = (TStringLiteralExpressionSyntax)context.Node;
+        var syntaxTree = context.Node.SyntaxTree;
+        var parseOptions = syntaxTree.Options;
+        if (this.CanConvertToInterpolatedString(stringLiteralExpression, parseOptions))
+        {
+            var diagnostic = Diagnostic.Create(CreateDescriptor(), stringLiteralExpression.GetLocation());
+            context.ReportDiagnostic(diagnostic);
+        }
+    }
+
+    protected abstract bool CanConvertToInterpolatedString(TStringLiteralExpressionSyntax stringLiteralExpression, ParseOptions parseOptions);
```

Comment:
Please move this abstract declaration somewhere to the top of the declaration. From my experience, it should be placed even before constructor, so it is immidiatelly obvious, what abstract things does this class have when reading from top to bottom. Maybe visit other similar places to verify, that mys suggested location is consistent with what is already there in the codebase